lib/ask_web/controllers/short_link_controller.ex

```diff
@@ -9,29 +9,34 @@ defmodule AskWeb.ShortLinkController do
       ShortLink
       |> Repo.get_by(hash: hash)
 
-    conn =
-      conn
-      |> assign(:skip_auth, true)
+    if link do
+      conn =
+        conn
+        |> assign(:skip_auth, true)
 
-    {path, query_string} =
-      case :binary.split(link.target, "?", [:global]) do
-        [path, query_string] -> {path, query_string}
-        [path] -> {path, ""}
-      end
+      {path, query_string} =
+        case :binary.split(link.target, "?", [:global]) do
+          [path, query_string] -> {path, query_string}
+          [path] -> {path, ""}
+        end
 
-    conn = %{
-      conn
-      | request_path: path,
-        path_info: split_path(path),
-        params: %{},
-        path_params: %{},
-        private: %{},
-        req_headers: [],
-        query_string: query_string,
-        query_params: %Plug.Conn.Unfetched{aspect: :query_params}
-    }
+      conn = %{
+        conn
+        | request_path: path,
+          path_info: split_path(path),
+          params: %{},
+          path_params: %{},
+          private: %{},
+          req_headers: [],
+          query_string: query_string,
+          query_params: %Plug.Conn.Unfetched{aspect: :query_params}
+      }
 
-    AskWeb.Endpoint.call(conn, [])
+      AskWeb.Endpoint.call(conn, [])
+    else
+      conn
+      |> send_resp(:not_found, "")
```

Comment:
```suggestion
      |> send_resp(:not_found, "Link not found")
```

So we show _some_ error to the user, other than the HTTP status code.